zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1011,15 +1012,29 @@ public static List<ACL> fixupACL(String path, List<Id> authInfo, List<ACL> acls)
             if (id == null || id.getScheme() == null) {
                 throw new KeeperException.InvalidACLException(path);
             }
-            if (id.getScheme().equals("world") && id.getId().equals("anyone") && !X509AuthenticationConfig
-                .getInstance().isX509ClientIdAsAclEnabled()) {
+            boolean isX509ZnodeGroupAclSuperUser =
```

Comment:
Can this logic be simplified? This boolean contains the results of `X509AuthenticationConfig.getInstance().isX509ClientIdAsAclEnabled()`. And in the `if` statement, we're checking yet again for `!X509AuthenticationConfig.getInstance().isX509ClientIdAsAclEnabled())`. As a result, this logic is confusing and verbose.

What is exactly meant by `isX509ZnodeGroupAclSuperUser`? Are we checking if it's `super`? or is it a cross-domain component? How does `isX509ClientIdAsAclEnabled` have anything to do with this?

Reply:
Reverted this logic, and currently checking in the next if-else block. Please check the updated code.